docs/usage/configuration-options.md

```diff
@@ -2969,6 +2969,34 @@ The above will set a replaceStrategy for any npm package which starts with `@ang
 
 The above will group together any npm package which starts with the string `angular`.
 
+### matchQl
+
+`matchQl` is an experimental query language which be used to create both simple and complex matching logic for `packageRules`.
+
+It can be used to create matching on fields not supported by other `matchX` fields (e.g. `"matchQl": "groupName = 'foo'"`) or also as an alternative for existing fields.
+
+Principles:
+
+- strings must be surrounded in single `'` or double `"` quotes
+- numbers, boolean values, `null` and `undefined` should not use quotes
+- arrays can only contain strings
+- `string = string` (e.g. `manager = "npm"` or `packageName = "@foo/*"`) will use pattern matching (glob, regex, or exact string matching)
+- `string == string` (e.g. `manager == "npm"`) will use exact string matching only
+- `string = array[string]` (e.g. `manager = ["npm", "dockerfile"]`) will return true if _any_ of the array values pattern match
+- `string != string` e.g. (`manager != "npm"`) returns true if it doesn't pattern match the string
+- `string != array[string]` (e.g. `manager != ["npm", "dockerfile"]`) will return true if _none_ of the array values pattern match
+- `number` values (e.g. `currentVersionAge`, which is measured in days) support comparisons `=`, `!=`, `<`, `<=`, `>` and `>=`
```

Comment:
```suggestion
- `number` values support comparisons `=`, `!=`, `<`, `<=`, `>` and `>=`, this comparison can be helpful for fields like `currentVersionAge` (which is measured in days)
```

Let's put the comparison first, and _then_ show the example. And expand on the example a bit.